provision.sh

```diff
@@ -230,10 +237,18 @@ ssl_ca_key: |
 $INDENTED_SSL_CA_KEY
 EOF
 
+# If the host is RHEL we'll need credentials to pull images
+# from Red Hat Container Image Registry
+if $SSH_CMD test -f /etc/redhat-release; then
```

Comment:
This test is not correct since CentOS also to have this file:
```
[centos@manila-cloud ~]$ cat /etc/redhat-release 
CentOS Stream release 8
```

I suggest you grep for the `Red Hat` inside the `/etc/redhat-release` file:
```suggestion
if $SSH_CMD grep -q "Red\ Hat" /etc/redhat-release; then
```